src/esp/io/JsonStlTypes.h

```diff
@@ -53,10 +53,8 @@ void addMember(JsonGenericValue& value,
                rapidjson::GenericStringRef<char> name,
                const std::vector<T>& vec,
                JsonAllocator& allocator) {
-  if (!vec.empty()) {
-    addMember(value, name, toJsonArrayHelper(vec.data(), vec.size(), allocator),
-              allocator);
-  }
+  addMember(value, name, toJsonArrayHelper(vec.data(), vec.size(), allocator),
```

Comment:
I know I kinda suggested this earlier, but now I'm having second thoughts. See this discussion here: https://stackoverflow.com/questions/9619852/what-is-the-convention-in-json-for-empty-vs-null

Let's discuss more over VC soon.

Reply:
As discussed, I reverted the last commit so that a `lightsChanged` flag is inserted when changes occur. This is simpler and safer.